jupyter_telemetry/eventlog.py

```diff
@@ -112,21 +108,66 @@ def register_schema(self, schema):
         jsonschema.validators.validator_for(schema).check_schema(schema)
 
         # Check that the properties we require are present
-        required_schema_fields = {'$id', 'version'}
+        required_schema_fields = {'$id', 'version', 'properties'}
         for rsf in required_schema_fields:
             if rsf not in schema:
                 raise ValueError(
                     '{} is required in schema specification'.format(rsf)
                 )
 
-        # Make sure reserved, auto-added fields are not in schema
-        if any([p.startswith('__') for p in schema['properties']]):
-            raise ValueError(
-                'Schema {} has properties beginning with __, which is not allowed'
-            )
+        for p, attrs in schema['properties'].items():
+            if p.startswith('__'):
+                raise ValueError(
+                    'Schema {} has properties beginning with __, which is not allowed'
+                )
+
+            # Validate "categories" property in proposed schema.
+            try:
+                cats = attrs['categories']
+                # Categories must be a list.
+                if not isinstance(cats, list):
+                    raise ValueError(
+                        'The "categories" field in a registered schemas must be a list.'
+                    )
+
+                # Unrestricted is a special case and must be listed alone:
+                if 'unrestricted' in cats and len(cats) > 1:
```

Comment:
Relax this constraint. Schema authors should be able to list other categories with `unrestricted`.